### DISCUSSION TITLE
How to crossref tbl?

### DISCUSSION THREAD

**Discussion by annezheng2** [quarto-dev/quarto-cli#6252] [Q&A]
### Description

I’m having an issue where I am unable to crossref tables. And I am getting the error: “WARNING: Unable to resolve crossref @tbl-test” when I try to render a PDF. The image generates as below but “?@tbl-test” appears rather than “Table 1”. 

```
@tbl-test
![Test Caption](Text/test.tex){#tbl-test} 
```

![image](https://github.com/A2-ai/DemoAnne/assets/138500255/5b9976da-23d7-455e-8c80-c6f793bf537b)

Ideally, it should function like Figure does below:
  
```
@fig-test
![Test Caption](Text/test.tex){#fig-test}
```
![edit](https://github.com/quarto-dev/quarto-cli/assets/138500255/737a640b-94ce-4087-9d86-f8803a4bb23a)

I tried using png and pdf as well and get the same error 
```
@tbl-test
![Test Caption](Text/test.pdf){#tbl-test}
```

**Comment by mcanouil** [comment]
`![Test Caption](Text/test.tex)` is the syntax for images, not for tables.
And `.tex` is definitely not a image extension.
I don't know what you try to do exactly.

Could you share a small self-contained "working" (reproducible) example to work with, _i.e._, a complete Quarto document or a Git repository? Thanks.

You can share a Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
---

This is a reproducible Quarto document using `format: html`.
It is written in Markdown and contains embedded R code.
When you run the code, it will produce a plot.

```{r}
plot(cars)
```

The end.
````
`````


This being said, with the upcoming redesign of the cross-reference system (watch #4944) you are going to be able to do weird thing such as referencing a table as a video, a theorem as a table, etc.
